### PR TITLE
Merges Azure simulations.

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

See ticket: https://github.com/obscuronet/obscuro-internal/issues/485.

We have two redundant ways to initialise an Azure enclave sim - one enclave, and multiple enclaves. But the former is just a special case of the latter, and doesn't need its own code path.

### What changes were made as part of this PR:

* Refactoring PR
* Removed the `OneAzureEnclave` sim
* Some additional checks to avoid case where too few/too many Azure enclaves are requested

### What are the key areas to look at

N/A

### CODE REVIEW

File: integration/simulation/network/azureenclave.go [hunk 1]
```diff
@@ -31,19 +33,14 @@ type networkWithAzureEnclaves struct {
...
-
 func NewNetworkWithAzureEnclaves(enclaveAddresses []string) Network {
+	if len(enclaveAddresses) == 0 {
+		panic("Cannot create azure enclaves network without at least one enclave address.")
+	}
 	return &networkWithAzureEnclaves{enclaveAddresses: enclaveAddresses}
 }
 
 func (n *networkWithAzureEnclaves) Create(params *params.SimParams, stats *stats.Stats) ([]ethclient.EthClient, []*obscuroclient.Client, []string, error) {
-	if len(n.enclaveAddresses) == 0 {
```
Comment (joeldudley) [inline integration/simulation/network/azureenclave.go:43]: I moved this check up to the constructor.

File: integration/simulation/network/azureenclave.go [hunk 2]
```diff
@@ -81,11 +78,11 @@ func (n *networkWithAzureEnclaves) Create(params *params.SimParams, stats *stats
 	// set up nodes with mock enclaves
 	for i := len(n.enclaveAddresses); i < params.NumberOfNodes; i++ {
 		// create a remote enclave server
-		enclavePort := uint64(params.StartPort + DefaultWsPortOffset + i)
-		enclaveAddress := fmt.Sprintf("%s:%d", Localhost, enclavePort)
+		nonAzureEnclavePort := uint64(params.StartPort + DefaultWsPortOffset + i)
```
Comment (joeldudley) [inline integration/simulation/network/azureenclave.go:81]: These two variables were conflicting with existing variables.

File: integration/simulation/simulation_azure_enclaves_test.go [hunk 3]
```diff
@@ -45,10 +50,14 @@ func TestOneAzureEnclaveNodesMonteCarloSimulation(t *testing.T) {
 	simParams.AvgNetworkLatency = simParams.AvgBlockDuration / 15
 	simParams.AvgGossipPeriod = simParams.AvgBlockDuration / 3
 
+	if len(vmIPs) > simParams.NumberOfNodes {
```
Comment (joeldudley) [inline integration/simulation/simulation_azure_enclaves_test.go:53]: Previously, there was a warning in a TODO to not accidentally do this. I replaced it with a proper check.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM thanks

**Pull request merged**